### PR TITLE
Model plot fix

### PR DESCRIPTION
**Pull request by sibirrer**
This pull request should fix bug #380 for at least a single band. A new routine to extract the 'amp' parameters from the kwargs_list was required.

### CODE REVIEW

**Comment by sibirrer** [comment]
@ajshajib I still need to add some test functions. But let me know if you see issues/concerns in the meantime

**Comment by sibirrer** [comment]
ok, @ajshajib , ready for review :)

File: lenstronomy/Plots/model_plot.py [hunk 1]
```diff
@@ -51,6 +51,18 @@ def __init__(self, multi_band_list, kwargs_model, kwargs_params, image_likelihoo
                                                        image_likelihood_mask_list=image_likelihood_mask_list)
 
         model, error_map, cov_param, param = self._imageModel.image_linear_solve(inv_bool=True, **kwargs_params)
```
Comment (ajshajib) [inline lenstronomy/Plots/model_plot.py:53]: Lines 53-56 needs to be changed like the code below to make the new changes to work.

```
        kwargs_params_copy = copy.deepcopy(kwargs_params)
        model, error_map, cov_param, param = self._imageModel.image_linear_solve(inv_bool=True, **kwargs_params)

        if linear_solver is False:
            if len(multi_band_list) > 1:
                raise ValueError('plotting the solution without the linear solver currently only works with one band.')

            kwargs_params = kwargs_params_copy
```

File: test/test_Plots/test_model_plot.py [hunk 2]
```diff
@@ -214,6 +214,19 @@ def test_check_solver_error(self):
...
+        kwargs_data = sim_util.data_configure_simple(numPix=10, deltaPix=1, background_rms=1, exposure_time=1)
+        # kwargs_data['image_data'] = np.zeros((10, 10))
+        kwargs_model = {'source_light_model_list': ['GAUSSIAN']}
+        kwargs_params = {'kwargs_lens': [],
+                         'kwargs_source': [{'amp': 2, 'sigma': 1, 'center_x': 0, 'center_y': 0}],
+                         'kwargs_ps': [], 'kwargs_lens_light': []}
+        lensPlot = ModelPlot(multi_band_list=[[kwargs_data, {'psf_type': 'NONE'}, {}]],
+                             kwargs_model=kwargs_model, kwargs_params=kwargs_params, bands_compute=[True],
+                             arrow_size=0.02, cmap_string="gist_heat", linear_solver=False)
+        lensPlot.plot_main(with_caustics=True)
```
Comment (ajshajib) [inline test/test_Plots/test_model_plot.py:227]: I suggest checking that `amp` keyword(s) within `kwargs_params` are staying the same as the initial.

**Review by ajshajib — CHANGES_REQUESTED** [review]
Hi @sibirrer, thanks a lot for implementing the fix! In my test, it's still not fully working, which I think can be solved by using `deepcopy`  as I suggested.

File: lenstronomy/Plots/model_plot.py (hunk 1 above)
Comment (sibirrer) [reply lenstronomy/Plots/model_plot.py]: ah, great catch! I made it slightly different such that the 'amp' parameters are updated if linear_solver=True but it's not happening if =False

**Review by ajshajib — APPROVED** [review]

**Comment by ajshajib** [comment]
Great! Thanks for implementing the requested changes, feel free to merge with the main branch. 🙂

**Pull request merged**